paint/data/dataset.py

```diff
@@ -0,0 +1,437 @@
+            mappings.CALIBRATION_FLUX_CENTERED_IMAGE_KEY: mappings.CALIBRATION_FLUX_CENTERED_IMAGE_IDENTIFIER,
+            mappings.CALIBRATION_PROPERTIES_KEY: mappings.CALIBRATION_PROPERTIES_IDENTIFIER,
+        }
+        return mapper[key]
+
+    @classmethod
+    def from_benchmark(
+        cls,
+        benchmark_file: Union[str, Path],
+        root_dir: Union[str, Path],
+        item_type: str,
+        download: bool = False,
+    ) -> tuple[Self, Self, Self]:
+        """
+        Initialize calibration dataset from a benchmark file.
+
+        This function returns a train, test, and validation dataset given a benchmark.
+
+        Parameters
+        ----------
+        benchmark_file : Union[str, Path]
+            Path to the file containing the benchmark information.
+        root_dir : Union[str, Path]
+            Directory where the dataset will be stored.
+        item_type : str
+            Type of item being loaded, i.e. raw image, cropped image, flux image, or calibration properties.
+        download : bool
+            Whether to download the data (Default: ``False``).
+
+        Returns
+        -------
+        PaintCalibrationDataset
+            Train dataset.
+        PaintCalibrationDataset
+            Test dataset.
+        PaintCalibrationDataset
+            Validation dataset.
+        """
+        root_dir = Path(root_dir)
+        log.info(
+            f"Begining the process of generating benchmark datasets. The file used to generate the benchmarks is:\n"
+            f" {benchmark_file}!"
+        )
+        # Load the splits data.
+        splits = pd.read_csv(benchmark_file)
+
+        # Check whether to download the data or not.
+        if download:
+            log.info("Downloading the benchmark items...")
+            split_ids = cls._download_benchmark_splits(
+                splits=splits, root_dir=root_dir, item_type=item_type
+            )
+        else:
+            # Check if the folder exists, if not the data must be downloaded.
+            if not root_dir.is_dir():
+                log.warning(
+                    "The root directory does not exist, the data has not yet been downloaded!\n"
+                    "The data will now be downloaded..."
+                )
+                split_ids = cls._download_benchmark_splits(
+                    splits=splits, root_dir=root_dir, item_type=item_type
+                )
+            # If the data is present, we can load it directly.
+            else:
+                split_ids = None
+
+        # Initialize a train dataset.
+        train_dataset = cls(
+            root_dir=Path(root_dir) / mappings.TRAIN_INDEX,
+            item_ids=split_ids[mappings.TRAIN_INDEX] if split_ids is not None else None,
+            item_type=item_type,
+        )
+
+        # Initialize a test dataset.
+        test_dataset = cls(
+            root_dir=Path(root_dir) / mappings.TEST_INDEX,
```

Comment:
root_dir is initialized multiple times as a pathlib.Path here. Possibly just do it once?